docs/pages/router/reference/testing.mdx

```diff
@@ -0,0 +1,100 @@
+
+## Configuration
+
+Before you proceed, ensure you've set up `jest-expo` according to the [Unit Testing documentation](/develop/testing) and [`@testing-library/react-native`](https://callstack.github.io/react-native-testing-library/docs/getting-started).
+
+## renderRouter
+
+`renderRouter` extends the functionality of [`render`](https://callstack.github.io/react-native-testing-library/docs/api#render) to simplify testing with Expo Router. It returns the the same query object as [`render`](https://callstack.github.io/react-native-testing-library/docs/api#render), and is compatible with [`screen`](https://callstack.github.io/react-native-testing-library/docs/api#screen), allowing you to use the standard [query API](https://callstack.github.io/react-native-testing-library/docs/api-queries) to locate components.
+
+`renderRouter` accepts the same [options](https://callstack.github.io/react-native-testing-library/docs/api#render-options) as `render` and introduces an additional option `initialRoute`, which sets an initial route for simulating deep-linking.
+
+### Inline file-system mock
+
+`renderRouter(mock: Record<string, ReactComponent>, options: RenderOptions)`
+
+`renderRouter` can provide inline-mocking of a file system by passing an object to this function as the first parameter. The keys of the object are the mock filesystem paths. Importantly, when defining these paths, do not use leading relative (`./`) or absolute (`/`) notation and exclude file extension.
+
+```tsx app.test.tsx
+import { renderRouter, screen } from 'expo-router/testing-library';
+
+it('my-test', async () => {
+  const MockComponent = jest.fn(() => <View />);
+
+  renderRouter(
+    {
+      index: MockComponent,
+      'folder/a': MockComponent,
+      '(group)/b': MockComponent,
+    },
+    {
+      initialRoute: '/folder/a',
+    }
+  );
+
+  expect(screen).toHavePathname('/folder/a');
+});
+```
+
+### Path to a directory
+
+`renderRouter(fixturePath: string, options: RenderOptions)`
+
+`renderRouter` can accept a directory path to mock an existing fixture. Ensure that the provided path is relative to the current test file.
+
+```tsx app.test.js
+it('my-test', async () => {
+  const MockComponent = jest.fn(() => <View />);
+  renderRouter('./my-test-fixture');
+});
+```
+
+### Path to a directory, with overrides
+
+`renderRouter({ appDir: string, overrides: Record<string, ReactComponent>}, options: RenderOptions)`
+
+For more intricate testing scenarios, `renderRouter` can leverage both directory path and inline-mocking methods simultaneously. The `appDir` parameter takes a string representing a pathname to a folder. The overrides parameter is an inline mock that can be used to override specific paths within the appDir. This combination allows for fine-tuned control over the mock environment.
+
+## Jest matchers
+
+The following matches have been added to `expect` and can be used to asset values on `screen`
+
+### `toHavePathname()`
```

Comment:
```suggestion
<APIBox header="toHavePathname()">
```
Let's use APIBox here for better readability. 

![CleanShot 2023-08-24 at 23 39 35](https://github.com/expo/expo/assets/10234615/c52b3d9b-3d55-4d33-87a3-8cee24f78327)